### DISCUSSION TITLE
Delta create_checkpoint() breaks Synapse SQL OPENROWSET

### DISCUSSION THREAD

**Discussion by Trebla2010** [delta-io/delta-rs#1611] [Q&A]
After running create_checkpoint(), I'm unable to run OPENROWSET. Getting below error.
I'm running this on Azure Synapse Serverless SQL.

SELECT
TOP 100 *
FROM
OPENROWSET(
BULK 'https://devstoragereporting.dfs.core.windows.net/enriched/TestData/',
FORMAT = 'DELTA'
) AS [result]

**Resolving Delta logs on path 'ADLSProcessing/staging/TestData/' failed with error: Cannot parse field field 'parts' in json object.**

I found that the file "_last_checkpoint" that gets created into the delta_log folder is what's breaking Synapse SQL. Fortunately, deleting this file restores functionality to my delta table and I keep the performance gain.
What is the purpose of the _last_checkpoint file? Is there a way to prevent this file from being generated?

Below is the content inside the _last_checkpoint file.
{"num_of_add_files":null,"parts":null,"size":132,"size_in_bytes":77024,"version":89}

**Comment by rtyler** [comment]
@Trebla2010 unfortunately this _seems_ like it might be the Synapse SQL delta integration having a bug in how it's handling the [Delta protocol](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#last-checkpoint-file)

The `_last_checkpoint` file is an integral part of the protocol, and the `parts` field is optional, from the linked document:

> The number of fragments if the last checkpoint was written in multiple parts. This field is optional.


Sorry if this isn't the answer you were hoping for [but our princess is in another castle](https://knowyourmeme.com/memes/but-our-princess-is-in-another-castle)
